packages/ag-charts-community/src/scene/group.ts

```diff
@@ -131,261 +62,175 @@ export class Group extends Node {
+        if (!this.clipRect) return;
 
-        for (const n of nodes) {
-            if (skipInvisible && (!n.visible || n.transitionOut)) continue;
+        // clipRect is in the group's coordinate space
+        const { x, y, width, height } = this.clipRect;
+        const { ctx } = renderCtx;
 
-            const bbox = n.getBBox();
+        ctx.save();
+        ctx.beginPath();
+        ctx.rect(x, y, width, height);
+        ctx.clip();
 
-            if (!bbox) continue;
+        this._debug?.(() => ({
+            name: this.opts?.name,
+            clipRect: this.clipRect,
+            ctxTransform: ctx.getTransform(),
+            renderCtx,
+            group: this,
+        }));
 
-            const { x, y, width, height } = bbox;
+        // clipBBox is in the canvas coordinate space,
+        // when we hit a layer we apply the new clipping
+        // at which point there are no transforms in play
+        return Transformable.toCanvas(this, this.clipRect);
+    }
 
-            if (x < left) {
-                left = x;
-            }
-            if (y < top) {
-                top = y;
-            }
-            if (x + width > right) {
-                right = x + width;
-            }
-            if (y + height > bottom) {
-                bottom = y + height;
+    protected renderChildren(children: Iterable<Node>, renderCtx: RenderContext) {
+        const { ctx, forceRender, stats } = renderCtx;
+        for (const child of children) {
+            // Skip invisible children, but make sure their dirty flag is reset.
+            if (!child.visible || !this.visible) {
+                child.markClean();
+                if (stats) {
+                    stats.nodesSkipped += nodeCount(child).count;
+                }
+                continue;
             }
-        }
 
-        return new BBox(left, top, right - left, bottom - top);
-    }
+            // Skip children that don't need to be redrawn.
+            if (!forceRender && child.dirty === RedrawType.NONE) {
+                if (stats) {
+                    stats.nodesSkipped += nodeCount(child).count;
+                }
+                continue;
+            }
 
-    setClipRect(bbox?: BBox) {
-        this.clipRect = bbox;
+            // Render marks this node (and children) as clean - no need to explicitly markClean().
+            ctx.save();
+            child.render(renderCtx);
+            ctx.restore();
+        }
     }
 
     /**
      * Transforms bbox given in the canvas coordinate space to bbox in this group's coordinate space and
      * sets this group's clipRect to the transformed bbox.
      * @param bbox clipRect bbox in the canvas coordinate space.
+     * @param useGroupCoordinateSpace keep provided bbox coordinate space.
      */
-    setClipRectInGroupCoordinateSpace(bbox?: BBox) {
-        this.clipRect = bbox ? Transformable.fromCanvas(this, bbox) : undefined;
+    setClipRect(bbox?: BBox, useGroupCoordinateSpace = true) {
```

Comment:
Nit: I think the naming is off here - `this.clipRect` should always be in local coordinate space, so `useGroupCoordinateSpace` is mis-named, it should actually be about the coordinate space of the `bbox` argument. E.g. `bboxInLocalCoordinateSpace` seems more representative?